compiler/plc_diagnostics/src/diagnostician.rs

```diff
@@ -147,27 +151,62 @@ impl DiagnosticAssessor for DefaultDiagnosticAssessor {
         match d {
             // improvements become warnings
             Diagnostic::ImprovementSuggestion { .. } => Severity::Warning,
+            _ if *d.get_type() == ErrNo::reference__unresolved => Severity::Critical,
             // everything else becomes an error
             _ => Severity::Error,
         }
     }
 }
 
 /// a diagnostics severity
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub enum Severity {
+    Critical,
     Error,
     Warning,
-    _Info,
+    Info,
 }
 
 impl std::fmt::Display for Severity {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let severity = match self {
+            Severity::Critical => "critical",
             Severity::Error => "error",
             Severity::Warning => "warning",
-            Severity::_Info => "info",
+            Severity::Info => "info",
         };
         write!(f, "{severity}")
     }
 }
+
+impl PartialOrd for Severity {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Severity {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (Severity::Critical, Severity::Critical) => std::cmp::Ordering::Equal,
+            (Severity::Critical, _) => std::cmp::Ordering::Greater,
+            (_, Severity::Critical) => std::cmp::Ordering::Less,
+            (Severity::Error, Severity::Error) => std::cmp::Ordering::Equal,
+            (Severity::Error, _) => std::cmp::Ordering::Greater,
+            (_, Severity::Error) => std::cmp::Ordering::Less,
+            (Severity::Warning, Severity::Warning) => std::cmp::Ordering::Equal,
+            (Severity::Warning, _) => std::cmp::Ordering::Greater,
+            (_, Severity::Warning) => std::cmp::Ordering::Less,
+            (Severity::Info, Severity::Info) => std::cmp::Ordering::Equal,
+        }
+    }
+}
+
+impl Severity {
+    pub fn combine(self, other: Self) -> Self {
```

Comment:
this is the same as Severity::max inherited from the Ord trait?

Reply:
yes i guess you're right